### DISCUSSION TITLE
eCAL TCP Layer, attempt to resolve .local instead of just hostname [Link-local connection]

### DISCUSSION THREAD

**Discussion by chengguizi** [eclipse-ecal/ecal#1521] [Ideas]
I understand that in the current TCP mode, in order to use eCAL properly, we have to manually key in the hostname-IP mapping on the subscriber machine. This is pretty inconvenient for deployment to customers' machine.

I have tested on both Windows and Ubuntu Linux hosts. There is a good workaround potentially to make the TCP mode work without config.

The trick is that, althought hostnames are not resolved direclty, espically when there is not present of a router (e.g. direct ethernet cable connect of the host and edge machine), by adding the ".local" domain, the mDNS mechanism kicks and systems are able to resolve the hostname.

My suggestion is then:

1. Make changes to the TCP pubsub behaviour, or at least make as a config setting, to fallback to .local domain, if the direct hostname could not be resolved
2. This is actually the behaviour of "ping" command in Windows. Which is pretty neat.

**Comment by FlorianReimold** [comment]
I converted this discussion to a [ticket](https://github.com/eclipse-ecal/ecal/issues/1531). I think this should work fine for most use cases. There are a few cases that I need to think about.

**Comment by chengguizi** [comment]
Much appreciated!
